Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -89,12 +94,19 @@ void CodeDiffDialog::CreateWidgets()
   m_matching_results_table->setColumnWidth(2, 4);
   m_matching_results_table->setColumnWidth(3, 210);
   m_matching_results_table->setColumnWidth(4, 65);
+  m_save_btn = new QPushButton(tr("Save"));
+  m_save_btn->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+  m_load_btn = new QPushButton(tr("Load"));
+  m_load_btn->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
   m_reset_btn = new QPushButton(tr("Reset All"));
   m_reset_btn->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
   m_help_btn = new QPushButton(tr("Help"));
   m_help_btn->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
   auto* help_reset_layout = new QHBoxLayout;
   help_reset_layout->addWidget(m_reset_btn, 0, Qt::AlignLeft);
+  help_reset_layout->addStretch();
+  help_reset_layout->addWidget(m_save_btn, 0, Qt::AlignRight);

```

Comment:
1. The Save button will need to take into account if the current list is empty. 
2. It should stay enabled even if the game is closed, so long as the list is not empty.

1. The Load button should only be enabled if the game is running.
2. If no file is found, a pop up message should warn the user no file was found.

You can reference how we allow the record button for the game running state.

Reply:
I'm not sure about saving while the game is closed. It needs to get the running game's ID to save.  I'd have to cache it and update it whenever a game is started or the window is opened.

Reply:
I think its fine to disable the Save button while emulation is not running.